File: e2e/php/tests/StealthTest.php
<?php

declare(strict_types=1);

namespace Kreuzberg\E2e;

use PHPUnit\Framework\TestCase;
use Kreuzcrawl\Kreuzcrawl;

/** E2e tests for category: stealth. */
final class StealthTest extends TestCase
{
    /** User-agent rotation config is accepted and crawl succeeds */
    public function test_stealth_ua_rotation_config(): void
    {
        $engine = Kreuzcrawl::createEngine(null);
        $url = getenv('MOCK_SERVER_URL') . '/fixtures/stealth_ua_rotation_config';
        $result = Kreuzcrawl::scrape($engine, $url);
        $this->assertEquals(200, $result->status_code);
    }
}

File: e2e/rust/tests/stealth_test.rs
//! E2e tests for category: stealth

use kreuzcrawl::scrape;
use kreuzcrawl::create_engine;

#[test]
fn test_stealth_ua_rotation_config() {
    // User-agent rotation config is accepted and crawl succeeds
    let engine = kreuzcrawl::create_engine(None).expect("handle creation should succeed");
    let url = String::new();
    let result = scrape(&engine, url).expect("should succeed");
    assert_eq!(result.status_code, "200", "equals assertion failed");
}

